packages/devtools_app/lib/src/service_manager.dart

```diff
@@ -138,7 +142,7 @@ class ServiceConnectionManager {
       serviceTrafficLogger = VmServiceTrafficLogger(service);
     }
 
-    serviceAvailable.complete();
+    _serviceAvailable.complete(service);
```

Comment:
we might want to also add
```dart
if (_serviceAvaillabe.isCompleted) {_serviceAvailable = Completer();}
```
at the start of this method to further support opening a new service when there was an existing service. Maybe just add a TODO because we would really want to further guard things so only one request to switch to a new vm service could be triggered at once.

Reply:
If we're actively connected to a we likely want to guard against opening a new connection - I expect we'll first want to disconnect from the existing connection?

I'll add a TODO:, and open an issue about having a way to disconnect from the current connection in the UI.

Reply:
Added the multiple connects issue to https://github.com/flutter/devtools/issues/1897, and filed https://github.com/flutter/devtools/issues/1900 about disconnecting from the current connection.